modules/auth/saas/settings/subscriptions.tsx

```diff
@@ -6,12 +6,103 @@ import Spinner from "@/resources/containers/spinner";
 import Service from "@/core/module/service.factory";
 import { useRouter } from "next/router";
 import moment from "moment";
+import Table from "@/resources/table";
 
 import { FaMoneyBillTransfer } from "react-icons/fa6";
 
+const data: TableDataModel[] = [
+  {
+    id: "Invoice 1",
+    date: new Date().toLocaleDateString(),
+    plan: "Basic",
+    amount: "€ 200",
+  },
+  {
+    id: "Invoice 2",
+    date: new Date().toLocaleDateString(),
+    plan: "Basic",
+    amount: "€ 150",
+  },
+];
+
+[3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20].forEach(
+  (i) => {
+    data.push({
+      id: `Invoice ${i}`,
+      date: new Date().toLocaleDateString(),
+      plan: "Basic",
+      amount: "€ 150",
+    });
+  }
+);
+
+const PaymentHistory: React.FC = () => {
+  const columns: ColumnConfig<TableDataModel>[] = [
+    { title: "Invoice ID", dataIndex: "id", sorter: true },
+    {
+      title: "Date",
+      render: (item) => <span>{item.date}</span>,
+    },
+    {
+      title: "Plan",
+      render: (item) => <span>{item.plan}</span>,
+      sorter: true,
+    },
+    { title: "Amount", dataIndex: "amount" },
+    {
+      title: "Actions",
+      render: () => (
+        <>
+          <a href="#">Download</a> | <a href="#">View</a>
+        </>
+      ),
+    },
+  ];
+
+  const itemsPerPage = 5;
+  const totalRecords = data.length;
+  const totalPages = Math.ceil(totalRecords / itemsPerPage);
+  const [currentPage, setCurrentPage] = useState(1);
+  const intialData = data.slice(
+    (currentPage - 1) * itemsPerPage,
+    currentPage * itemsPerPage
+  );
+  const [paginatedData, setPaginatedData] =
+    useState<TableDataModel[]>(intialData);
+
+  useEffect(() => {
+    setPaginatedData(() => {
+      return data.slice(
+        (currentPage - 1) * itemsPerPage,
+        currentPage * itemsPerPage
+      );
+    });
+  }, [currentPage]);
```

Comment:
Can this logic go in the table component? Is this a server or a backend pagination?